frame/support/procedural/src/pallet/parse/call.rs

```diff
@@ -233,4 +233,15 @@ impl CallDef {
 			docs: helper::get_doc_literals(&item.attrs),
 		})
 	}
+
+	pub fn empty(attr_span: proc_macro2::Span) -> Self {
+		Self {
+			where_clause: None,
+			instances: Vec::new(),
+			index: 0,
```

Comment:
same here index is None, I think we should not fake parsing an item, but we should expand correctly from proper non existing call.

Reply:
maybe do like the hook code or change index to `Option<usize>` or make the the expand code generate a new item in `Def::item` so you have a proper CallDef